staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go

```diff
@@ -59,17 +59,17 @@ var buildDefinitions sync.Once
 var namer *openapi.DefinitionNamer
 
 // BuildSwagger builds swagger for the given crd in the given version
-func BuildSwagger(crd *apiextensions.CustomResourceDefinition, version string) (*spec.Swagger, error) {
+func BuildSwagger(crd *apiextensions.CustomResourceDefinition, oldStructuralSchema *structuralschema.Structural, version string) (*spec.Swagger, *structuralschema.Structural, error) {
```

Comment:
needing to pass an old schema into this method doesn't make sense to me. if there are invalid types in the schema, we should not publish openapi for it

Reply:
yes for openapi publishing we can stop publishing invalid schema and it's [backwards compatible](https://github.com/kubernetes/kubernetes/pull/79587#issuecomment-507462802)